src/Compilers/Shared/BuildClient.cs

```diff
@@ -219,6 +217,13 @@ private int RunLocalCompilation(string[] arguments, BuildPaths buildPaths, TextW
                 return null;
             }
 
+            // A temporary directory is required for the server to startup hence we bail out early
```

Comment:
Is this a condition we will need to be able to diagnose in the wild? If so, should something be logged or thrown here?

It's not clear to me under what conditions the client can fail to get a temp path, and whether we should just keep going with a client-side compilation in that case. It feels like if this happens, it will just manifest as a mysterious increase in solution build time for the end user.

Reply:
I took @cston suggestion to add an error here. 

This is the kind of situation where it should never happen but technically it can if you really screw with environment variables so you have to handle that outcome.